web/src/user_groups.ts

```diff
@@ -284,6 +284,28 @@ export function is_user_in_group(user_group_id: number, user_id: number): boolea
     return false;
 }
 
+export function is_user_in_setting_group(
+    setting_group: GroupSettingType,
+    user_id: number,
+): boolean {
+    if (typeof setting_group === "number") {
+        return is_user_in_group(setting_group, user_id);
+    }
+
+    const direct_members = setting_group.direct_members;
+    if (direct_members.includes(user_id)) {
+        return true;
+    }
+
+    const direct_subgroups = setting_group.direct_subgroups;
+    for (const direct_subgroup_id of direct_subgroups) {
+        if (is_user_in_group(direct_subgroup_id, user_id)) {
+            return true;
+        }
+    }
+    return false;
+}
```

Comment:
This probably isn't important, but the set-based algorithm using `get_recursive_subgroups` would be more efficient if there were a lot of direct subgroups that shared subgroups.